frontend/src/Hooks/Hooks.tsx

```diff
@@ -2,14 +2,95 @@ import { useToast } from "@chakra-ui/react";
 import { STATS_PATH } from "../Config/Paths";
 import { GPUStats } from "../Data";
 import { useAuth } from "../Providers/AuthProvider";
-import { fire, success, validatedElim, Validation } from "../Utils/Utils";
+import {
+  fire,
+  success,
+  Validated,
+  validatedElim,
+  Validation,
+} from "../Utils/Utils";
 import { API_URL } from "../App";
 import { useJarJar } from "../Utils/Hooks";
 import { useInterval } from "@chakra-ui/react";
 
 const ADD_MACHINE_URL = "/add_workstation";
 const REMOVE_MACHINE_URL = "/rm_workstation";
 
+const GET_ALL_FILES_URL = "/list_files";
+const UPLOAD_FILE_URL = "/attach_file";
+const REMOVE_FILE_URL = "/remove_file";
+const GET_SPECIFIC_FILE_URL = "/get_file";
+
+export const useGetAllFiles = (
+  hostname: string,
+  callback: (r: Validated<Response>) => void,
+) => {
+  const { useAuthFetch } = useAuth();
+  const [, getAllFilesAuth] = useAuthFetch(
+    GET_ALL_FILES_URL + "?hostname=" + hostname,
+    callback,
+  );
+
+  return () => getAllFilesAuth({ method: "GET" });
+};
+
+const encodeFile = (file: Uint8Array) => {
+  let binaryString = "";
+  const len = file.byteLength;
+  for (let i = 0; i < len; i++) {
+    binaryString += String.fromCharCode(file[i]);
+  }
+  return btoa(binaryString);
+};
+
+export const useUploadFile = (callback: (r: Validated<Response>) => void) => {
+  const { useAuthFetch } = useAuth();
+  const [, uploadFileAuth] = useAuthFetch(UPLOAD_FILE_URL, callback);
```

Comment:
Could add toast here further down in the callback, or here in the hook. *maybe* higher-up here makes sense? (so if called elsewhere we don't have to duplicate toast logic)